### PR TITLE
Automatically Odyssey versioning when merging to main

### PR DESCRIPTION
**Pull request by KevinGhadyani-Okta**
https://oktainc.atlassian.net/browse/OKTA-663938

## Summary

We need to version Odyssey when merging to `main` so we can automate publishing of Odyssey altogether. Right now, it's a very manual process, and this is the first step in fixing that.

Once we have this fixed, we can move onto automating the publish process and then the post-publish tasks.

## Testing & Screenshots
N/A

### CODE REVIEW

File: .github/workflows/version.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+      - name: node_modules cache
+        uses: actions/cache@v2
+        with:
+          path: "**/node_modules"
+          key: ${{ runner.os }}-node-modules-cache-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: ${{ runner.os }}-node-modules-cache-
+
+      - name: Install dependencies
+        run: yarn install --immutable
```
Comment (chrispulsinelli-okta) [inline .github/workflows/version.yml:31]: Should we have a check after this to make sure that `yarn.lock` was not modified?

File: .github/workflows/version.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        run: |
+          git add yarn.lock
+          git add '*.json'
+          git commit -m "build: bump versions"
+
+      - name: Commit CHANGELOG changes
+        run: |
+          yarn prettier-write
+          git add '*.md'
+          git commit -m "docs: update CHANGELOGs"
```
Comment (chrispulsinelli-okta) [inline .github/workflows/version.yml:46]: Do you need the lowercase "s" or are you attempting to pluralize "CHANGELOG"?

File: .github/workflows/version.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+      - name: Push changes
+        run: git push origin main
```
Comment (chrispulsinelli-okta) [inline .github/workflows/version.yml:49]: Any concerns of this token being unable to do this next year? I recall hearing about certain permissions being removed from some tokens

**Review by chrispulsinelli-okta — COMMENTED** [review]
Changes look good to me overall. Left some questions and comments for you to consider.

File: .github/workflows/version.yml (hunk 1 above)
Comment (KevinGhadyani-Okta) [reply .github/workflows/version.yml:31]: That's what `--immutable` does. It's like `npm install --ci`.

File: .github/workflows/version.yml (hunk 2 above)
Comment (KevinGhadyani-Okta) [reply .github/workflows/version.yml:46]: It updates `CHANGELOG` in every package and the root package, so it's multiple of them.

File: .github/workflows/version.yml (hunk 3 above)
Comment (KevinGhadyani-Okta) [reply .github/workflows/version.yml:49]: This is the token in the repo as far as I understand. Without the token, I don't think it'd be able to run any of these tasks.

And if it stops working next year, we can deal with it then. I'm more concerned about it not working this year :P.

**Review by chrispulsinelli-okta — APPROVED** [review]

**Pull request merged**